### PR TITLE
Refresh download instructions for new multi-repo world

### PR DESCRIPTION
**Pull request by directhex**
Plz don't merge until after d15.1 is stable, but I'd appreciate a review.

This obsoletes https://github.com/mono/website/pull/259

### CODE REVIEW

File: docs/getting-started/install/linux/index.md
Comment (akoeplinger) [inline docs/getting-started/install/linux/index.md]: This needs to be a dependency of mono-devel in the next release so I don't think we need to mention it explicitly.

File: docs/getting-started/install/linux/index.md
Comment (akoeplinger) [inline docs/getting-started/install/linux/index.md]: I'd like something similar to https://www.microsoft.com/net/core#linuxredhat where you choose your distro first so the page is smaller.

Should be easily doable given we already use such a switcher on https://github.com/mono/website/blob/gh-pages/download/index.html#L23, just horizontal.

File: docs/getting-started/install/linux/index.md
Comment (akoeplinger) [inline docs/getting-started/install/linux/index.md]: What's the status of Ubuntu 12.04? Can it continue to use the wheezy packages?

File: docs/getting-started/install/linux/index.md
Comment (akoeplinger) [inline docs/getting-started/install/linux/index.md]: repo/debian is correct here, instead of e.g. repo/ubuntu ?

File: docs/getting-started/install/linux/index.md
Comment (akoeplinger) [inline docs/getting-started/install/linux/index.md]: Does this mean we don't support openSUSE anymore?

File: docs/getting-started/install/linux/index.md
Comment (akoeplinger) [inline docs/getting-started/install/linux/index.md]: can't comment on the line: should we do something about the "Security updates for older releases" section below? does it still apply in the new world?

File: docs/getting-started/install/linux/index.md
Comment (akoeplinger) [inline docs/getting-started/install/linux/index.md]: we may want to consider splitting alpha/beta to their own page as it gets pretty long

File: docs/getting-started/install/linux/index.md
Comment (directhex) [reply docs/getting-started/install/linux/index.md]: I hate how much we're coming to depend on MSbuild while its build process is so flakey :/

File: docs/getting-started/install/linux/index.md
Comment (directhex) [reply docs/getting-started/install/linux/index.md]: Didn't even hit 1% of RPM users in the logs(!)

File: docs/getting-started/install/linux/index.md
Comment (directhex) [reply docs/getting-started/install/linux/index.md]: Yeah, that indicates a directory on the server. Best I could do is symlink ubuntu to debian so it looks like it's different

File: docs/getting-started/install/linux/index.md
Comment (directhex) [reply docs/getting-started/install/linux/index.md]: EOL in 28 days. Not worth thinking about any more.

File: docs/getting-started/install/linux/index.md
Comment (directhex) [reply docs/getting-started/install/linux/index.md]: It's still accurate for the named releases. We could ditch the section if we're disclaiming responsibility for the versions listed, but would need to bring it back in future if any new vulns are found & we offer security updates to old versions.

File: docs/getting-started/install/linux/index.md
Comment (directhex) [reply docs/getting-started/install/linux/index.md]: I agree.

File: docs/getting-started/install/linux/index.md
Comment (directhex) [reply docs/getting-started/install/linux/index.md]: Sounds nice. Any ideas on implementation? Is there existing CSS to handle a horizontal switcher?

Actually, if we go that route, can we punt it to the main Download page?

Comment (akoeplinger) [reply docs/getting-started/install/linux/index.md]: the line I linked to should have the skeleton that you need if you do `s/vertical/horizontal/`.

re: main Download page: I image it'd be too cramped there with the layout we have right now.

File: docs/getting-started/install/linux/index.md
Comment (akoeplinger) [reply docs/getting-started/install/linux/index.md]: It's still the default on Travis iirc 😞

File: docs/getting-started/install/linux/index.md
Comment (akoeplinger) [reply docs/getting-started/install/linux/index.md]: yeah a symlink sounds good to avoid any confusion.

File: docs/getting-started/install/linux/index.md
Comment (akoeplinger) [reply docs/getting-started/install/linux/index.md]: Same story as with roslyn. Maybe we should do the same and just check in the binaries for now?

Comment (akoeplinger) [reply docs/getting-started/install/linux/index.md]: Given that we need a story for people doing `./configure; make; make install` from the tarball that might be a good alternative because they'll be screwed otherwise once we fully deprecate xbuild (unless they somehow find out how to download+build msbuild)

File: docs/getting-started/install/linux/index.md
Comment (akoeplinger) [reply docs/getting-started/install/linux/index.md]: Hmm, maybe we should just change this and _not_ provide symlinks for snapshots from e.g. 3.10 -> 3.10.X.X so users need to always pin to a specific version. A security fix would then just be "pin to a different version".

File: docs/getting-started/install/linux/index.md
Comment (directhex) [reply docs/getting-started/install/linux/index.md]: Travis has 27 days to fix it before it becomes insecure, then

File: docs/getting-started/install/linux/index.md
Comment (directhex) [reply docs/getting-started/install/linux/index.md]: OK, symlink in place. Also a symlink for alpha-wheezy -> alpha, so consistent structure is available. Should I change the docs to match?

File: docs/getting-started/install/linux/index.md
Comment (directhex) [reply docs/getting-started/install/linux/index.md]: I'd like to give it a shot. I just need a modern jekyll installed to test a local gh-pages branch, right?

Comment (akoeplinger) [reply docs/getting-started/install/linux/index.md]: Yes, just follow https://github.com/mono/website#setting-up-a-local-copy-of-the-website.

Newer Jekyll also supports `--incremental` so the whole site isn't rebuilt all the time which takes a while.

File: docs/getting-started/install/linux/index.md
Comment (akoeplinger) [reply docs/getting-started/install/linux/index.md]: Yep, please change the docs to be consistent.

File: docs/getting-started/install/linux/index.md
Comment (akoeplinger) [inline docs/getting-started/install/linux/index.md]: we should keep the old version of the stable install page for the initial alpha release.

File: download/alpha.html
Comment (akoeplinger) [inline download/alpha.html]: `Mono is available as an Apt repository for ` is redundant, remove.

File: download/alpha.html
Comment (akoeplinger) [inline download/alpha.html]: when I visit the page it shows all distro instructions and not just the preselected Debian ones. Is that intended or a bug?

File: download/alpha.html
Comment (akoeplinger) [inline download/alpha.html]: I wonder if we could just make this `echo "deb http://download.mono-project.com/repo/debian alpha-$(lsb_release -cs) main" ... `

This would allows us to get rid of the repeated instructions. Do you know if lsb_release is available in most configurations (I heard something about some distros moving away from it)?

File: download/alpha.html
Comment (akoeplinger) [inline download/alpha.html]: SVG instead of PNG (we can do it in a follow up cleanup if you don't have them handy).

**Review by akoeplinger — CHANGES_REQUESTED** [review]

File: download/alpha.html
Comment (directhex) [reply download/alpha.html]: I don't have these handy. I wasn't sure how best to do these, since we use some magic Javascript to do the Win/Mac/Lin icons.

File: download/alpha.html
Comment (directhex) [reply download/alpha.html]: That's a bug. An interesting one!

The lack of pre-selection is (I think) due to a lack of the pre-selection JS in the alpha version of the page ( https://github.com/mono/website/blob/gh-pages/download/index.html#L108-L124 ) but showing all of them is definitely a bug

File: download/alpha.html
Comment (directhex) [reply download/alpha.html]: Pushed a change for this one

File: download/alpha.html
Comment (directhex) [reply download/alpha.html]: I don't believe it's guaranteed to exist on any distro, and it's a bit harder for people using deriavtives like Mint where they might know which Ubuntu or Debian they're equivalent to, but not how to modify the script. Maybe. Maybe that's a hypothetical user.

Comment (akoeplinger) [reply download/alpha.html]: Hmm, that's a good point. Let's keep it as it is for now.

Though we could strip the "Add the Mono repository to your system:" and "Then update or install mono-devel to get started" from each section and move it to a more general one. Which reminds me, what happened to the other info shown on the stable download page such as "Usage" and "Notes"?

Comment (directhex) [reply download/alpha.html]: How much of the Usage/Notes there is useful/interesting?

Comment (akoeplinger) [reply download/alpha.html]: Most of it? I think we can shorten it a bit, I can come up with something. We can do this after merging.

Comment (directhex) [reply download/alpha.html]: There's stuff that still needs fixing up, but what is blocking merging right now?

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**
